Add Groupsize test

rcritten opened the pull request:
This test adds an LDIF of a specific number of users with all as members of the group allusers. The time to add one more user is recorded.

antoniotorresm commented on src/ipaperftest/core/constants.py:
```diff
@@ -545,3 +562,45 @@ def getLevelName(level):
+          [global]
+          debug=True
+    - command:
+        cmd: "systemctl restart httpd"
```
You could use the service module instead: https://docs.ansible.com/ansible/2.9/modules/service_module.html

antoniotorresm commented on src/ipaperftest/core/constants.py:
```diff
@@ -545,3 +562,45 @@ def getLevelName(level):
+    - package:
+        name: python3-pip
+    - package:
+        name: time
```
You can install multiple packages with a single call to the package module using a list instead: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_module.html

antoniotorresm commented on src/ipaperftest/core/main.py:
```diff
@@ -123,6 +123,8 @@ def run(self, ctx):
 @click.option("--ad-threads", default=0, help="Active Directory login threads "
                                               "to run per client during AuthenticationTest.")
+@click.option("--sizelimit", default=100, help="IPA search size limit")
+@click.option("--ad-threads", default=0, help="Active Directory login threads ")
```
`--ad-threads` is duplicated, I guess it's an error?

rcritten commented:
I think I've addressed the reported issues.

rcritten commented:
batch.py was not added to the repo but I'm not completely sure what I intended it to do. It shows users in a batch. I don't know why we would need that.

antoniotorresm commented:
Thanks, it looks good to me now. Since this is a new test it would be nice to have a section for it in the README.

rcritten commented:
- added support for sub groups
- added yield to capture times
- If no subgroups are provided then the original test runs
- Added --no-members option to group-add (perhaps controversial)
- rebased on top of other changes

antoniotorresm commented on create-test-data.py:
```diff
@@ -713,6 +739,8 @@ def put_entry(self, entry):
               is_flag=True)
 @click.option("--debug", default=False, help="Debug logging", is_flag=True)
+@click.option("--number-of-subgroups", default=0, help="Number of subgroups to create.",
+              type=int)
```
There's no need to specify the type if there's a default value.

rcritten replied:
removed

antoniotorresm commented on src/ipaperftest/core/plugin.py:
```diff
@@ -177,7 +177,8 @@ def generate_metadata(self, ctx):
 
         for c in self.generate_clients(ctx):
-            machine_configs.append(c)
+            if c:
```
What's the reason for this?

rcritten replied:
IIRC I was running with zero clients, none are necessary. But something was being returned and my being impatient I think this was my solution rather than having the generator return nothing. I'll take another look at it.

rcritten replied:
Right, so generate_clients() by default returns None so there is no machine_config to append, hence it blows up later in the code. yielding () instead has basically the same effect.

antoniotorresm reviewed (changes requested):
Could you add a section to README explaining this test? Also update the usage to show the new option.

antoniotorresm commented:
Could you update the output of `ipaperftest --help` in the README as well?

rcritten commented:
help section updated
